### PR TITLE
Hugbox

### PR DESCRIPTION
**Pull request by Ikarrus**
hugbox

You can hug it and maybe it will even hug you back.

Starts in perma, and as a rare drop in maint. Also as an emoji.

### CODE REVIEW

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 1]
```diff
@@ -616,4 +616,23 @@
...
+	new /obj/item/weapon/grenade/chem_grenade/metalfoam(src)
+
+
+/obj/item/weapon/storage/box/hug
+	name = "box of hugs"
+	desc = "A special box for sensitive people."
+	icon_state = "hugbox"
+	foldable = null
+
+/obj/item/weapon/storage/box/attack_self(mob/user)
```
Comment (GunHog) [inline code/game/objects/items/weapons/storage/boxes.dm]: Yay! I can now HUG ALL BOXES!!

**Comment by Nienhaus** [comment]
Wait, you can actually hug it? Amazing. Should port everywhere.

**Comment by freerealestate** [comment]
Just seems like coder passive-aggressiveness towards the players.

**Comment by xxalpha** [comment]
No it's a sign of affection. The box has a heart on it.

**Comment by bgobandit** [comment]
this isn't what a hugbox looks like

look how meta that comment was

**Comment by Cheridan** [comment]
@freerealestate Is this a sensitive topic for you? I have a box for you that can help with that.

**Comment by bustygoku** [comment]
/TG/STATION AND THE SUBSEQUENT CODEBASE ARE #SAFESPACES PLEASE NO MORE BULLYING OF THE INNOCENT PLAYERS!

**Comment by freerealestate** [comment]
You act like I care about this much more than I actually do. This is just a
symbol of coder smugness and a self-indulgence.

**Comment by Miauw62** [comment]
The hugbox will never threaten to stab you. In fact, the hugbox cannot speak.

**Comment by BestName** [comment]
Why not an actual hugbox?
hug-machine-32669-1238773329-10

**Comment by Ikarrus** [comment]
Too literal.

**Comment by Dorsisdwarf** [comment]
This is amazing and wonderful, and should be given to admins as an option button in ahelps for when someone gets mad at them.

**Comment by MrStonedOne** [comment]
Inb4 valve sues us because they think it's a companion cube ripoff

**Comment by Kelenius** [comment]
@freerealestate stop being a cunt, k, thx.

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 2]
```diff
@@ -616,4 +616,24 @@
...
+	name = "box of hugs"
+	desc = "A special box for sensitive people."
+	icon_state = "hugbox"
+	foldable = null
+
+/obj/item/weapon/storage/box/hug/attack_self(mob/user)
+	..()
+	user.changeNext_move(CLICK_CD_MELEE)
+	playsound(loc, "rustle", 50, 1, -5)
+	user.visible_message("<span class='notice'>[user] hugs the [src].</span>","<span class='notice'>You hug the [src].</span>")
```
Comment (Kelenius) [inline code/game/objects/items/weapons/storage/boxes.dm]: >Not actually making a prob(1) && prob(10) chance of it hugging you back

**Comment by Ikarrus** [comment]
@MrStonedOne They should also sue us for Viscerators and Stun Batons

**Comment by GunHog** [comment]
I want to shill for getting the HL2 voice files back. Also yes for having it hug you back!

**Comment by Dorsisdwarf** [comment]
I want to shill for more fucking voice files and trimming the unneeded ones. We don't have "because", "so", "or" and "in order", but we have "johnson", "abortions", "freeman","ass" and "motherfucker"?

File: code/game/objects/items/weapons/storage/boxes.dm (hunk 2 above)
Comment (Incoming5643) [inline code/game/objects/items/weapons/storage/boxes.dm]: prob(prob(prob(1))) for a proper one in a million chance

**Comment by Incoming5643** [comment]
If you remove ass blast USA I will never forgive you

**Comment by GunHog** [comment]
I did not mean to derail the PR discussion! Keep it to hugs and boxes! And hugs IN boxes!

**Comment by Ricotez** [comment]
Does it speak? And if it does, am I encouraged by Nanotrasen to ignore it?

**Comment by bustygoku** [comment]
It should have an incredibly low chance to whisper some really demonic shit into your ear when you hug it.

**Comment by PKPenguin321** [comment]
definitely needs an adminbutton that spawns it on a player if they make a whiny adminhelp :+1:

**Comment by Xhuis** [comment]
thumbsup

**Comment by Dorsisdwarf** [comment]
Player response on bagil: Very positive

**Pull request merged**